flink-scala/src/test/scala/org/apache/flink/api/scala/typeutils/EnumValueSerializerCompatibilityTest.scala

```diff
@@ -19,26 +19,25 @@ package org.apache.flink.api.scala.typeutils
 
 import org.apache.flink.api.common.typeutils.{TypeSerializerSchemaCompatibility, TypeSerializerSnapshotSerializationUtil}
 import org.apache.flink.core.memory.{DataInputViewStreamWrapper, DataOutputViewStreamWrapper}
-import org.apache.flink.util.TestLogger
 
-import org.junit.{Rule, Test}
-import org.junit.Assert._
-import org.junit.rules.TemporaryFolder
-import org.scalatest.junit.JUnitSuiteLike
+import org.assertj.core.api.Assertions.{assertThat, fail}
+import org.junit.Test
+import org.junit.jupiter.api.io.TempDir
 
 import java.io._
 import java.net.{URL, URLClassLoader}
+import java.nio.file.Files
 
 import scala.reflect.NameTransformer
 import scala.tools.nsc.{Global, Settings}
 import scala.tools.nsc.reporters.ConsoleReporter
 
-class EnumValueSerializerCompatibilityTest extends TestLogger with JUnitSuiteLike {
+class EnumValueSerializerCompatibilityTest {
 
-  private val _tempFolder = new TemporaryFolder()
+  private val _tempFolder = Files.createTempDirectory("")
 
-  @Rule
-  def tempFolder = _tempFolder
+  @TempDir
+  private def tempFolder = _tempFolder
```

Comment:
Is `@TempDir` utilized in any way like this? IMO we could leverage `@TempDir` as a test method param approach here so it is not necessary to deal with the required initialization:
```java
  @Test
  def checkIdenticalEnums(@TempDir tempFolder: Path): Unit = {
    assertThat(checkCompatibility(enumA, enumA, tempFolder).isCompatibleAsIs).isTrue
  }
```

It has to be added to every method, but IMO it will be cleaner overall. Or maybe init the class level field with `TestFileUtils.createTempDir()`, which will be cleaned up.